utils/math.py

```diff
@@ -29,3 +37,357 @@ def get_rounded_mean_safe(cls, values):
         if (len(values) > 0):
             return round(mean(values), cls.nb_decimal_numbers)
         return 0
+    
+    @classmethod
+    def weighted_matrix(cls, matrix: np.ndarray, weights: list) -> np.ndarray:
+        """
+        Returns a weighted matrix obtained by multiplying each column of the input matrix
+        by the corresponding weight from the provided list.
+
+        Args:
+            matrix (np.ndarray): The input matrix to be weighted.
+            weights (list): A list of weights to multiply each column of the matrix.
+
+        Returns:
+            np.ndarray: The resulting weighted matrix.
+
+        """
+        if len(weights) != matrix.shape[1]:
+            raise ValueError("Weights must be the same length as the number of column of the given matrix.")
+        
+        weighted_matrix = matrix.copy()
+
+        for index, weight in enumerate(weights):
+            weighted_matrix[:, index] *= weight
+
+        return weighted_matrix
+
+    @classmethod
+    def calculte_matrix_extremum(self, matrix: np.ndarray, impacts: list) -> np.ndarray:
```

Comment:
This one should have stayed within the TOPSIS class as it's quite specific to TOPSIS